consensus/istanbul/core/core.go

```diff
@@ -197,6 +197,17 @@ func (c *core) startNewRound(round *big.Int) {
 		logger = c.logger.New("old_round", c.current.Round(), "old_seq", c.current.Sequence())
 	}
 
+	logger.Trace("Start new ibft round")
+
+	// If new round is 0, then check if qbftConsensus needs to be enabled
+	if round.Uint64() == 0 && c.backend.IsQBFTConsensus() {
```

Comment:
Why is the switch from IBFT to QBFT triggered here? Should it not be triggered by in `backend.Commit()`? So concerns are well separated between `ibft` and `qbft`?